datajoint_utilities/dj_worker/worker.py

```diff
@@ -294,14 +306,12 @@ def _purge_invalid_jobs(JobTable, table):
     if not jobs_query:
         return
 
-    invalid_count = len(jobs_query) - len(table.key_source)
     invalid_removed = 0
-    if invalid_count > 0:
-        for key, job_key in zip(*jobs_query.fetch("KEY", "key")):
-            if not (table.key_source & job_key):
-                (jobs_query & key).delete()
-                invalid_removed += 1
-
-        logger.info(
-            f"{invalid_removed}/{invalid_count} invalid jobs removed for `{dj.utils.to_camel_case(table.table_name)}`"
-        )
+    for key, job_key in zip(*jobs_query.fetch("KEY", "key")):
+        if (not (table.key_source & job_key)) or (table & job_key):
```

Comment:
Does this have lower performance? It definitely reads better.
```suggestion
        if job_key not in table.key_source or job_key in table:
```

Reply:
yes it does

`table.key_source & job_key` is much faster than `job_key in table.key_source`

Reply:
This does not try to be transaction safe. Is this only used in low-activity scenarios?

Reply:
I see. Because this is just the jobs table, which is not critical for data consistency.